### ISSUE TITLE
Function parameter can either access scope- or const-npc-variables

### ISSUE THREAD

**Issue by Jeybla** [rathena/rathena#426]
Example Script:
```
prontera,162,221,5	script	ScopeTest	899,{
	set .npc,1;
	set .@scope,2;
	set @char,3;
	callfunc("func",.npc,.@scope,@char);
	callfunc("func",.@scope,.npc,@char);
	callfunc("func",@char,.@scope,.npc);
	callfunc("func",@char,.npc,.@scope);
	end;
}
function	script	func	{
	dispbottom "arg1: "+getarg(0)+" arg2: "+getarg(1)+" arg3: "+getarg(2);
	return;
}
```

Output:
"arg1: 1 arg2: 0 arg3: 3"
"arg1: 2 arg2: 0 arg3: 3"
"arg1: 3 arg2: 2 arg3: 0"
"arg1: 3 arg2: 1 arg3: 0"

The character variable does not have impact on it.

**Comment by Jeybla** [comment]
This diff works for me:
```
 src/map/script.c | 22 ++++++++++++++++------
 1 file changed, 16 insertions(+), 6 deletions(-)

diff --git a/src/map/script.c b/src/map/script.c
index 5026c74..3a5667c 100644
--- a/src/map/script.c
+++ b/src/map/script.c
@@ -5035,7 +5035,8 @@ static int menu_countoptions(const char* str, int max_count, int* total)
 	struct script_retinfo* ri;
 	struct script_code* scr;
 	const char* str = script_getstr(st,2);
-	DBMap **ref = NULL;
+	DBMap **ref_npc = NULL;
+	DBMap **ref_scope = NULL;
 
 	scr = (struct script_code*)strdb_get(userfunc_db, str);
 	if( !scr )
@@ -5052,11 +5053,20 @@ static int menu_countoptions(const char* str, int max_count, int* total)
 		{
 			const char* name = reference_getname(data);
 			if( name[0] == '.' ) {
-				if(!ref){
-					ref = (struct DBMap**)aCalloc(sizeof(struct DBMap*), 1);
-					ref[0] = (name[1] == '@' ? st->stack->var_function : st->script->script_vars);
+				if( name[1] == '@' ) {
+					if(!ref_scope) {
+						ref_scope = (struct DBMap**)aCalloc(sizeof(struct DBMap*), 1);
+						ref_scope[0] = st->stack->var_function;
+					}
+					data->ref = ref_scope;
+				}
+				else {
+					if(!ref_npc) {
+						ref_npc = (struct DBMap**)aCalloc(sizeof(struct DBMap*), 1);
+						ref_npc[0] = st->script->script_vars;
+					}
+					data->ref = ref_npc;
 				}
-				data->ref = ref;
 			}
 		}
 	}
@@ -5067,7 +5077,7 @@ static int menu_countoptions(const char* str, int max_count, int* total)
 	ri->pos          = st->pos;// script location
 	ri->nargs        = j;// argument count
 	ri->defsp        = st->stack->defsp;// default stack pointer
-	push_retinfo(st->stack, ri, ref);
+	push_retinfo(st->stack, ri, ref_npc);
 
 	st->pos = 0;
 	st->script = scr;
```

I know it's not an important fix. But it may prevent someones headache :>

**Comment by Atemo** [comment]
Issue fixed in 6295c77d393555fe5a366f092a75f4cf0d2de933 script engine upgrade
